source/extensions/filters/http/composite/filter.cc

```diff
@@ -27,6 +30,24 @@ template <class... Ts> struct Overloaded : Ts... { using Ts::operator()...; };
 template <class... Ts> Overloaded(Ts...) -> Overloaded<Ts...>;
 
 } // namespace
+
+std::unique_ptr<ProtobufWkt::Struct> MatchedActionInfoType::buildProtoStruct() const {
+  auto message = std::make_unique<ProtobufWkt::Struct>();
+  auto& fields = *message->mutable_fields();
+  for (const auto& p : actions) {
+    fields[p.first] = ValueUtil::stringValue(p.second);
+  }
+  return message;
+}
+
+absl::optional<std::string> MatchedActionInfoType::serializeAsString() const {
+  return Json::Factory::loadFromProtobufStruct(*buildProtoStruct().get())->asJsonString();
+}
+
+void MatchedActionInfoType::setFilterAction(const std::string& filter, const std::string& action) {
+  actions[filter] = action;
+}
+
```

Comment:
nit: the definition of  `setFilterAction` and `serializeAsString` function can just be added to the header file, like what you already did for `serializeAsProto`, as they are 1 line function body. This is a bit more consistent I think